validator_client/rest_api/src/validator.rs

```diff
@@ -0,0 +1,262 @@
+    req: Request<Body>,
+    validator_store: Arc<ValidatorStore<T, E>>,
+) -> BoxFut {
+    let response_builder = ResponseBuilder::new(&req);
+    let future = req
+        .into_body()
+        .concat2()
+        .map_err(|e| ApiError::ServerError(format!("Unable to get request body: {:?}", e)))
+        .and_then(|chunks| {
+            serde_json::from_slice::<ValidatorRequest>(&chunks).map_err(|e| {
+                ApiError::BadRequest(format!(
+                    "Unable to parse JSON into ValidatorRequest: {:?}",
+                    e
+                ))
+            })
+        })
+        .and_then(move |body| {
+            let validator_pubkey = body.validator;
+            validator_store
+                .set_validator_status(&validator_pubkey, true)
+                .ok_or_else(|| ApiError::ServerError("Validator pubkey not present".into()))
+        })
+        .and_then(|_| response_builder?.body_empty());
+    Box::new(future)
+}
+
+/// Stops proposing/attesting for the given validator.
+/// The validator must already be known by the validator client.
+pub fn stop_validator<T: SlotClock + 'static, E: EthSpec>(
+    req: Request<Body>,
+    validator_store: Arc<ValidatorStore<T, E>>,
+) -> BoxFut {
+    let response_builder = ResponseBuilder::new(&req);
+    let future = req
+        .into_body()
+        .concat2()
+        .map_err(|e| ApiError::ServerError(format!("Unable to get request body: {:?}", e)))
+        .and_then(|chunks| {
+            serde_json::from_slice::<ValidatorRequest>(&chunks).map_err(|e| {
+                ApiError::BadRequest(format!(
+                    "Unable to parse JSON into ValidatorRequest: {:?}",
+                    e
+                ))
+            })
+        })
+        .and_then(move |body| {
+            let validator_pubkey = body.validator;
+            validator_store
+                .set_validator_status(&validator_pubkey, false)
+                .ok_or_else(|| ApiError::ServerError("Validator pubkey not present".into()))
+        })
+        .and_then(|_| response_builder?.body_empty());
+    Box::new(future)
+}
+
+/// Generates a `VoluntaryExit` message for a given validator and
+/// publishes it to the network.
+pub fn exit_validator<T: SlotClock + 'static, E: EthSpec>(
+    req: Request<Body>,
+    validator_store: Arc<ValidatorStore<T, E>>,
+    beacon_node: Arc<RemoteBeaconNode<E>>,
+) -> BoxFut {
+    let response_builder = ResponseBuilder::new(&req);
+    let bn1 = beacon_node.clone();
+    let future = req
+        .into_body()
+        .concat2()
+        .map_err(|e| ApiError::ServerError(format!("Unable to get request body: {:?}", e)))
+        .and_then(|chunks| {
+            serde_json::from_slice::<ValidatorRequest>(&chunks).map_err(|e| {
+                ApiError::BadRequest(format!(
+                    "Unable to parse JSON into ValidatorRequest: {:?}",
+                    e
+                ))
+            })
+        })
+        .and_then(move |body| {
```

Comment:
should we get an unsigned `VoluntaryExit` from beacon node api instead of constructing it here in the validator http api?